client/src/app/site/pages/meetings/pages/agenda/modules/topics/services/topic-pdf.service/topic-pdf.service.ts

```diff
@@ -0,0 +1,183 @@
+        private pollPercentBase: PollPercentBasePipe,
+        private pollService: TopicPollService
+    ) {}
+
+    /**
+     * Generates an pdf out of a given topic and saves it as file
+     *
+     * @param topic the topic to export
+     */
+    public exportSingleTopic(topic: ViewTopic): void {
+        const doc = this.topicToDocDef(topic);
+        const filename = `${this.translate.instant(`Topic`)}_${topic.title}`;
+        const metadata = {
+            title: filename
+        };
+
+        this.pdfDocumentService.download({ docDefinition: doc, filename, metadata });
+    }
+
+    /**
+     * Main function to control the pdf generation.
+     * Calls all other functions to generate the PDF in multiple steps
+     *
+     * @param topic the ViewTopic to create the document for
+     * @returns a pdfmake compatible document as document
+     */
+    private topicToDocDef(topic: ViewTopic): object {
+        const title = this.createTitle(topic);
+        const text = this.createTextContent(topic);
+        const pollResult = this.createPollResultTable(topic.polls);
+
+        return [title, text, pollResult];
+    }
+
+    /**
+     * Creates the title for PDF
+     * TODO: Cleanup. Should be reused from time to time. Can be in another service
+     *
+     * @param topic the ViewTopic to create the document for
+     * @returns the title part of the document
+     */
+    private createTitle(topic: ViewTopic): object {
+        return {
+            text: topic.title,
+            style: `title`
+        };
+    }
+
+    /**
+     * Creates the  description part of the document. Also converts the parts of an topic to PDF
+     *
+     * @param topic the ViewTopic to create the document for
+     * @returns the description of the topic
+     */
+    private createTextContent(topic: ViewTopic): object {
+        if (topic.text) {
+            return this.htmlToPdfService.addPlainText(topic.text);
+        } else {
+            return {};
+        }
+    }
+
+    /**
+     * Creates the poll result table for all published polls
+     * TODO: Make this reusable (also used in assignment-pdf.service)
+     *
+     * @param polls the ViewPoll objects to create the document for
+     * @returns the table as pdfmake object
+     */
+    private createPollResultTable(polls: ViewPoll[]): object {
+        const resultBody = [];
+        for (const poll of polls) {
+            if (poll.isPublished) {
+                const pollTableBody = [];
+
+                resultBody.push({
+                    text: poll.title,
+                    bold: true,
+                    style: `textItem`,
+                    margin: [0, 15, 0, 0]
+                });
+
+                pollTableBody.push([
+                    {
+                        text: ``,
+                        style: `tableHeader`
+                    },
+                    {
+                        text: this.translate.instant(`Candidates`),
```

Comment:
`Candidates` in a topic poll? Topic polls have text options so it should just be `Options`